### DISCUSSION TITLE
set_lreal from utili

### DISCUSSION THREAD

**Discussion by gborre** [gijzelaerr/python-snap7#427] [Q&A] [answered]
Hello!
I was wondering if I was misusing the set_lreal function or there's something not working. This is the current implementation.
`
def set_lreal(bytearray_: bytearray, byte_index: int, lreal: float) -> bytearray: 

    lreal = float(lreal)
    _bytes = struct.unpack('8B', struct.pack('>d', lreal))
    bytearray_[byte_index] = _bytes[0]
    return
 `
Using the function as it is I got wrong results writing on PLC. I did a small change to make it work: 
`
def set_lreal(bytearray_: bytearray, byte_index: int, lreal: float) -> bytearray: 

    lreal = float(lreal)
    _bytes = struct.unpack('8B', struct.pack('>d', lreal))
    **bytearray_[byte_index: byte_index + 8] = _bytes[:]**
    return
 `
Is this correct? Am I doing something wrong? 

Thanks!

**Comment by nikteliy** [comment, marked as answer]
Hi @gborre
I'm sorry for the delayed response. You're right, it's a bug. I've modified the code of the `lreal` functions in [this commit](https://github.com/gijzelaerr/python-snap7/pull/444/commits/53a1feedefec0770c64cdd6f63f38a718811f02c) Also, I added you as a co-author, hope you don't mind.
Thanks for your help

**Comment by gijzelaerr** [comment]
Closing as resolved. The `set_lreal` bug was fixed in PR #444.
